installers-conda/build_installers.py

```diff
@@ -0,0 +1,402 @@
+)
+p.add_argument(
+    "--artifact-name", action="store_true",
+    help="Print computed artifact name and exit.",
+)
+p.add_argument(
+    "--extra-specs", nargs="+", default=[],
+    help="One or more extra conda specs to add to the installer",
+)
+p.add_argument(
+    "--licenses", action="store_true",
+    help="Post-process licenses AFTER having built the installer. "
+    "This must be run as a separate step.",
+)
+p.add_argument(
+    "--images", action="store_true",
+    help="Generate background images from the logo (test only)",
+)
+p.add_argument(
+    "--cert-id", default=None,
+    help="Apple Developer ID Application certificate common name."
+)
+p.add_argument(
+    "--lite", action="store_true",
+    help=f"Do not include packages {scientific_packages.keys()}"
+)
+args = p.parse_args()
+
+yaml = YAML()
+yaml.indent(mapping=2, sequence=4, offset=2)
+indent4 = partial(indent, prefix="    ")
+
+SPYVER = get_version(SPYREPO).strip().split("+")[0]
+
+specs = {
+    "spyder": "=" + SPYVER,
+    "paramiko": "",
+    "pyxdg": "",
+}
+if SPECS.exists():
+    logger.info(f"Reading specs from {SPECS}...")
+    _specs = yaml.load(SPECS.read_text())
+    specs.update(_specs)
+else:
+    logger.info(f"Did not read specs from {SPECS}")
+
+if not args.lite:
+    specs.update(scientific_packages)
+
+for spec in args.extra_specs:
+    k, *v = re.split('([<>= ]+)', spec)
+    specs[k] = "".join(v).strip()
+    if k == "spyder":
+        SPYVER = v[-1]
+
+OUTPUT_FILE = DIST / f"EXPERIMENTAL-{APP}-{SPYVER}-{OS}-{ARCH}.{EXT}"
+INSTALLER_DEFAULT_PATH_STEM = f"{APP}-{SPYVER}"
+
+
+def _generate_background_images(installer_type):
+    """Requires pillow"""
+    if installer_type == "sh":
+        # shell installers are text-based, no graphics
+        return
+
+    from PIL import Image
+
+    logo_path = SPYREPO / "img_src" / "spyder.png"
+    logo = Image.open(logo_path, "r")
+
+    if installer_type in ("exe", "all"):
+        sidebar = Image.new("RGBA", (164, 314), (0, 0, 0, 0))
+        sidebar.paste(logo.resize((101, 101)), (32, 180))
+        output = DIST / "spyder_164x314.png"
+        sidebar.save(output, format="png")
+
+        banner = Image.new("RGBA", (150, 57), (0, 0, 0, 0))
+        banner.paste(logo.resize((44, 44)), (8, 6))
+        output = DIST / "spyder_150x57.png"
+        banner.save(output, format="png")
+
+    if installer_type in ("pkg", "all"):
+        _logo = Image.new("RGBA", logo.size, "WHITE")
+        _logo.paste(logo, mask=logo)
+        background = Image.new("RGBA", (1227, 600), (0, 0, 0, 0))
+        background.paste(_logo.resize((148, 148)), (95, 418))
+        output = DIST / "spyder_1227x600.png"
+        background.save(output, format="png")
+
+
+def _get_condarc():
+    # we need defaults for tensorflow and others on windows only
+    defaults = "- defaults" if WINDOWS else ""
+    prompt = "[spyder]({default_env}) "
+    contents = dedent(
+        f"""
+        channels:  #!final
+          - spyder-ide
```

Comment:
Do we really need to have our channel here? I think there are no important packages there.